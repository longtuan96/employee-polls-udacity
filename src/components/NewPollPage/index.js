import React, { useState } from "react";
import "./NewPollPage.css";
import { useNavigate } from "react-router";
import { useDispatch, useSelector } from "react-redux";
import { selectAuthedUserId } from "../../redux/slices/authSlice";
import {
  fetchQuestionList,
  saveQuestion,
} from "../../redux/slices/questionSlice";

const NewPollPage = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const authedUserId = useSelector(selectAuthedUserId);
  const [firstOptionInput, setFirstOptionInput] = useState("");
  const [secondOptionInput, setSecondOptionInput] = useState("");

  const handleFirstOptionInputChange = (e) => {
    const value = e.target.value;
    setFirstOptionInput(value);
  };

  const handleSecondOptionInputChange = (e) => {
    const value = e.target.value;
    setSecondOptionInput(value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const payload = {
      optionOneText: firstOptionInput,
      optionTwoText: secondOptionInput,
      author: authedUserId,
    };
    dispatch(saveQuestion(payload));
    dispatch(fetchQuestionList());
    navigate("/");
  };
  return (
    <div>
      <h1 className="">Would You Rather</h1>
      <form onSubmit={handleSubmit}>
        <div className="">
          <label
            htmlFor="firstOption"
            data-testid="firstOptionLabel"
            className=""
          >
            First Option
          </label>
          <div className="">
            <input
              value={firstOptionInput}
              onChange={handleFirstOptionInputChange}
              type="text"
              name="firstOption"
              id="firstOption"
              data-testid="firstOption"
              className=""
            />
          </div>
        </div>

        <div className="">
          <label
            htmlFor="secondOption"
            data-testid="secondOptionLabel"
            className=""
          >
            Second Option
          </label>
          <div className="">
            <input
              value={secondOptionInput}
              onChange={handleSecondOptionInputChange}
              type="text"
              name="secondOption"
              id="secondOption"
              data-testid="secondOption"
              className=""
            />
          </div>
        </div>

        <div className="">
          <button type="submit" className="">
            Submit
          </button>
        </div>
      </form>
    </div>
  );
};

export default NewPollPage;
