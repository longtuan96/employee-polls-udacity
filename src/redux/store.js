import { configureStore } from "@reduxjs/toolkit";
import authReducer from "./slices/authSlice";
import userReducer from "./slices/userSlice";
import questionReducer from "./slices/questionSlice";
export const store = configureStore({
  reducer: {
    auth: authReducer,
    user: userReducer,
    question: questionReducer,
  },
});
