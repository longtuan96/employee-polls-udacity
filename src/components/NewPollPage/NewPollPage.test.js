import { fireEvent, render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import React from "react";
import NewPollPage from ".";
import { store } from "../../redux/store";

describe("NewPoll", () => {
  it("should render the component", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <NewPollPage />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });

  it("should display all required elements", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <NewPollPage />
        </BrowserRouter>
      </Provider>
    );

    const firstOptionInputElement = component.getByTestId("firstOptionInput");
    const secondOptionInputElement = component.getByTestId("secondOptionInput");
    const submitButton = component.getByTestId("submit-btn");
    expect(submitButton.textContent).toBe("Submit");

    fireEvent.change(firstOptionInputElement, {
      target: { value: "something 1" },
    });
    fireEvent.change(secondOptionInputElement, {
      target: { value: "something 2" },
    });
    expect(firstOptionInputElement.value).toBe("something 1");
    expect(secondOptionInputElement.value).toBe("something 2");
  });
});
