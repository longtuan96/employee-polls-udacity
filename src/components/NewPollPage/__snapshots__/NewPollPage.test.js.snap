// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPoll should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="new-poll-page-container"
      >
        <h1
          class=""
        >
          Would You Rather
        </h1>
        <form>
          <div
            class=""
          >
            <label
              class=""
              for="firstOption"
            >
              First Option
            </label>
            <div
              class=""
            >
              <input
                class=""
                data-testid="firstOptionInput"
                id="firstOption"
                name="firstOption"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class=""
          >
            <label
              class=""
              for="secondOption"
            >
              Second Option
            </label>
            <div
              class=""
            >
              <input
                class=""
                data-testid="secondOptionInput"
                id="secondOption"
                name="secondOption"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class=""
          >
            <button
              class=""
              data-testid="submit-btn"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="new-poll-page-container"
    >
      <h1
        class=""
      >
        Would You Rather
      </h1>
      <form>
        <div
          class=""
        >
          <label
            class=""
            for="firstOption"
          >
            First Option
          </label>
          <div
            class=""
          >
            <input
              class=""
              data-testid="firstOptionInput"
              id="firstOption"
              name="firstOption"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class=""
        >
          <label
            class=""
            for="secondOption"
          >
            Second Option
          </label>
          <div
            class=""
          >
            <input
              class=""
              data-testid="secondOptionInput"
              id="secondOption"
              name="secondOption"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class=""
        >
          <button
            class=""
            data-testid="submit-btn"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
